src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+
+/**
+ * Class to represent parsing the commands
+ */
+public class Parser {
+    private enum Commands { list, done, todo, event, deadline, delete, find }
+
+
+    /**
+     * Method that parses instruction from the user.
+     *
+     * @param instruction user's instruction
+     * @throws NoDescriptionError If user inputs nothing after a task command.
+     * @throws UnknownCommandError If user inputs a command that is outside the scope of the chatbot
+     */
+    public static String parse(String instruction, Ui ui, TaskList tasks) throws NoDescriptionError, UnknownCommandError {
+        String[] splitInstructions = instruction.split(" ", 2);
+        String operative = splitInstructions[0];
+        Commands command;
+        String[] temp;
+        String item;
+        String date;
+        String description;
+        String keyword;
+        Task toAdd;
+        int taskPointer;
```

Comment:
Take note of the below coding standard.

![image](https://user-images.githubusercontent.com/39460640/131533697-885885e0-75d5-4fc0-9ba4-27c4772faf6b.png)

Variables should be private as well, looks like no need to keep them package-private.